html/includes/graphs/application/powerwall_battery.inc.php

```diff
@@ -0,0 +1,24 @@
+
+
+<?php
+
+$scale_min = '0';
+
+$rrd_filename = rrd_name($device['hostname'], 'powerwall_battery');
```

Comment:
This is the wrong rrd name.  your rrds are named `app-powerwall-3.rrd` for example. Look at the other application based graphs to see how.

also, `var_dump($variable);` to see what a variable is set too. much better than guessing.

Reply:
No murrant is saying that your poller file is creating the file name like `$rrd_name = array('app', $name, $app_id);` which results in a file called `app-powerwall-3.rrd` so you need to build the same filename in the graph code as well otherwise the graphs won't work / show anything.

In general the `var_dump()` is a function you can use in testing to print out the variables for you to see what they are called so `var_dump($rrd_filename);` will print out what the filename is.

Reply:
Are you sure my poller application is even right to begin with? Most of it was guesswork

Reply:
Looks fine. Some optimisations need to be done but get it working first.

Reply:
This is literally what you need:
`$rrd_filename = rrd_name($device['hostname'], array('app', $app['app_type'], $app['app_id']));`

You can find that by looking at most of the other application graphs.

Reply:
I managed to get it working last night
https://snag.gy/li0gZE.jpg
https://snag.gy/KaWc57.jpg

I'll add the other parameters today and hopefully tidy it all up.

Reply:
I now have a basic system working, but feel there are too many graphs, so I want to cull some off first.
There are currently 17 graph files, which is excessive, and there are a couple that I do want to add, but they are COUNTER not GAUGE, so I'm not sure how to represent them (energy usage, so can hopefully see daily usage)

Also, trying to work out the multiple line graph, to make a 4 in 1 overview graph.